### PR TITLE
doc: update documents to reflect changes in test/mpi

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
The test suite can be configured and run independently. Update misc documents to reflect it.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,9 @@
+===============================================================================
+                               Changes in 4.1
+===============================================================================
+
+# Testsuite (test/mpi) are configured separately from mpich configure.
```
Comment (raffenet) [inline CHANGES:5]:
```suggestion
# Testsuite (test/mpi) is configured separately from mpich configure.
```

File: doc/installguide/install.tex.vin [hunk 2]
```diff
@@ -668,6 +668,15 @@ standard output, along with an XML version of the test results in
 top-level (\texttt{mpich}) directory will run tests of the commands, such as
 \texttt{mpicc} and \texttt{mpiexec}, that are included with MPICH.
 
+MPICH test suite can be used to test any MPI implementations, not just MPICH.
+To test a pre-installed MPI implementation, there is no need to configure and
+build MPICH. Go to \texttt{test/mpi}` directory and
```
Comment (raffenet) [inline doc/installguide/install.tex.vin:673]:
```suggestion
The MPICH test suite can be used to test any MPI implementations, not just MPICH.
To test a pre-installed MPI implementation, there is no need to configure and
build MPICH. Go to \texttt{test/mpi}` directory and
```

File: doc/installguide/install.tex.vin [hunk 3]
```diff
@@ -668,6 +668,15 @@ standard output, along with an XML version of the test results in
...
+
+\begin{verbatim}
+    configure --with-mpi=/path/to/mpi
```
Comment (raffenet) [inline doc/installguide/install.tex.vin:676]:
```suggestion
    ./configure --with-mpi=/path/to/mpi
```

**Review by raffenet — CHANGES_REQUESTED** [review]
A few minor suggested tweaks.

File: doc/installguide/install.tex.vin (hunk 3 above)
Comment (hzhou) [reply doc/installguide/install.tex.vin]: Thanks. All fixed.

**Review by raffenet — APPROVED** [review]

**Pull request merged**